src/audio/module_adapter/module_adapter.c

```diff
@@ -1319,6 +1405,12 @@ int module_adapter_bind(struct comp_dev *dev, void *data)
 	/* set the source dev pointer */
 	mod_source_info->sources[source_index] = source_dev;
 
+	/* Disable 1 source 1 sink simple copy mode if multiple source buffers */
+	if (module_adapter_multi_sink_source_check(dev)) {
+		comp_dbg(dev, "module_adapter_bind(), multi_sink_or_source = true");
+		mod->multi_source_or_sink = true;
```

Comment:
you can simplify this to mod->multi_source_or_sink = module_adapter_multi_sink_source_check(dev);